main.tf

```diff
@@ -158,23 +216,56 @@ resource "aws_security_group" "this" {
 }
 
 resource "aws_security_group_rule" "ingress" {
-  count                    = var.create_security_group ? 1 : 0
-  type                     = var.ingress_type
+  for_each                 = var.ingress_rules
+  type                     = "ingress"
   description              = "Allow inbound traffic from existing Security Groups"
-  from_port                = var.port
-  to_port                  = var.port
-  protocol                 = var.ingress_protocol
+  from_port                = lookup(each.value, "from_port")
+  to_port                  = lookup(each.value, "to_port")
+  protocol                 = "tcp"
   source_security_group_id = join("", aws_security_group.this.*.id)
   security_group_id        = join("", aws_security_group.this.*.id)
 }
 
 resource "aws_security_group_rule" "egress" {
-  count             = var.create_security_group ? 1 : 0
-  type              = var.egress_type
+  for_each          = var.egress_rules
+  type              = "egress"
   description       = "Allow all egress traffic"
-  from_port         = var.from_port
```

Comment:
For `description` since `from_port` and `to_port` are not predetermined, I think it would be nice to have a generic description like "Allow custom egress traffic".